### PR TITLE
Feature - Prevents multiple audio/video attachments from being played at the same time

### LINKED ISSUE #24010: Multiple audio/video attachments can be played at the same time
### Steps to reproduce the problem

1. Post two audio or video files or find an account that has two such posts
2. Press play on one attachment
3. Press play on another one

### Expected behaviour

When you play a media file while another one is already playing, the currently playing file should stop

### Actual behaviour

The files play simultaneously

### Detailed description

You can do 3 too, in fact you can do as many as would fit into your viewport + 1 more in the pop-out player.
Снимок_экрана_2023-03-06_в_23_18_43

### Specifications

Mastodon 4.1.0 (mastodon.social)
Safari 16.1 (17614.2.9.1.13, 17614) and Vivaldi 5.7.2921.60 (Stable channel) (arm64), macOS 12.6.1 but I doubt it depends on the browser

### PR DESCRIPTION
**Pull request by TheDevJoao**
## This PR fixes [#24010](https://github.com/mastodon/mastodon/issues/24010) 

- [x] pauses all other videos once a new video is played
- [x] pauses all other audio files once a new audio file is played
- [x] tests

### Preview
multiple_videos

multiple_audios

### CODE REVIEW

File: app/javascript/mastodon/features/audio/index.jsx [hunk 1]
```diff
@@ -22,6 +22,8 @@ const messages = defineMessages({
 const TICK_SIZE = 10;
 const PADDING   = 180;
 
+var currentAudio = null;
```
Comment (Gargron) [inline app/javascript/mastodon/features/audio/index.jsx:25]: Maybe it would make sense to have a `currentMedia` shared between audio/video components?

Comment (TheDevJoao) [reply app/javascript/mastodon/features/audio/index.jsx:25]: Good call! Changing this soon 🤝

File: app/javascript/mastodon/features/__tests__/toggle-play-test.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
```
Comment (nschonni) [inline app/javascript/mastodon/features/__tests__/toggle-play-test.jsx:1]: Aside: I don't think you need the name "test" in the file

Comment (TheDevJoao) [reply app/javascript/mastodon/features/__tests__/toggle-play-test.jsx:1]: @nschonni Nice, just commited the change now.

**Comment by TheDevJoao** [comment]
Hey fellas @Gargron @nschonni! Are any more changes required to merge this PR?

**Review by Gargron — DISMISSED** [review]

**Comment by TheDevJoao** [comment]
Fixed a merge conflict, requesting another review to merge this PR @Gargron.

**Review by Gargron — APPROVED** [review]

**Pull request merged**